src/tiff.ml

```diff
@@ -821,29 +826,33 @@ module Ifd = struct
     { entries; data_offsets; data_bytecounts; ro = reader; header }
 end
 
-type t = { header : header; ifd : Ifd.t }
+type ('repr, 'kind) kind =
+  | Uint8 : (int, Bigarray.int8_unsigned_elt) kind
+  | Int8 : (int, Bigarray.int8_signed_elt) kind
+  | Uint16 : (int, Bigarray.int16_unsigned_elt) kind
+  | Int16 : (int, Bigarray.int16_signed_elt) kind
+  | Int32 : (int32, Bigarray.int32_elt) kind
+  | Float32 : (float, Bigarray.float32_elt) kind
+  | Float64 : (float, Bigarray.float64_elt) kind
+
+type ('repr, 'kind) t = {
+  data_type : ('repr, 'kind) kind;
+  header : header;
+  ifd : Ifd.t;
+}
 
 let ifd t = t.ifd
 
-let from_file (f : File.ro) =
+let from_file (type a b) (f : File.ro) (data_type : (a, b) kind) : (a, b) t =
```

Comment:
```suggestion
let from_file (type a b) (data_type : (a, b) kind) (f : File.ro)  : (a, b) t =
```
I think the kind argument should go before the `ro`, I think it would be more likely to partially apply `from_file` with a `kind` to get a `from_file` for say `Uint8`s ?